### PR TITLE
Feature/expectations

### LINKED ISSUE #9: Support expectations
Add support for expectation processing. 

See http://www.mumuki.org/expectativas/plataform/2015/06/16/el-lenguaje-de-expectativas/

### PR DESCRIPTION
**Pull request by fedescarpa**
Fixes #9

### CODE REVIEW

File: server/expectation-checker.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var _ = require('lodash');
+var j = require('junify');
+
+var syntax = require('./syntax');
```
Comment (flbulgarelli) [inline server/expectation-checker.js:6]: Nice!

File: test/expectations.test.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    assert(!hasUsage('var foo;', 'foo', 'bar'));
+  });
+
+  it('when binding does not exist', function () {
+    assert(!hasUsage('5 + 2;', 'foo', 'bar'));
+  });
+
+});
+
+describe('utils', function () {
```
Comment (flbulgarelli) [inline test/expectations.test.js]: Perhaps we could move this test to another file

Comment (fedescarpa) [inline test/expectations.test.js]: ok

File: server/syntax.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    [{ type: 'VariableDeclaration', declarations: j.variable('sub'), _: j._ }, function (result) {
+      return result.sub.some(function (it) {
+        return expressionHasUsage(it, target);
+      });
+    }],
+    [{ type: 'VariableDeclarator', init: j.variable('sub'), _: j._ }, function (result) {
+      return expressionHasUsage(result.sub, target);
+    }],
+    [{ type: 'Identifier', name: j.variable('sub'), _: j._ }, function (result) {
+      return result.sub === target;
```
Comment (flbulgarelli) [inline server/syntax.js]: Here we could just match the expected name and return `_.constant(true)`, right?

File: server/syntax.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    if (match) {
+      return callback(match);
+    }
+  }
+  throw new Error('Pattern matchnig error ' + JSON.stringify(arg));
+}
+
+module.exports = {
+  match: match,
+  matches: matches,
```
Comment (flbulgarelli) [inline server/syntax.js]: Perhaps we could extract `match` and `matches`  to another module that only has extensions to `junify`. 

And we could also rename `match` to `matchesAny`, and `matches` to `match`\- those names reveal intention better.

File: server/syntax.js (hunk 3 above)
Comment (fedescarpa) [inline server/syntax.js]: yes, of course! I don't know why I commited this change. I was sure I had
it that way

File: server/syntax.js (hunk 4 above)
Comment (fedescarpa) [inline server/syntax.js]: done

**Pull request merged**